test/acceptance/filepermissions_test.go

```diff
@@ -0,0 +1,118 @@
+	"fmt"
+	"github.com/aws/amazon-cloudwatch-agent-test/environment"
+	"github.com/aws/amazon-cloudwatch-agent-test/filesystem"
+	"github.com/aws/amazon-cloudwatch-agent-test/internal/common"
+	"github.com/aws/amazon-cloudwatch-agent-test/test/status"
+	"github.com/stretchr/testify/assert"
+	"log"
+	"testing"
+	"time"
+)
+
+var envMetaDataStrings = &(environment.MetaDataStrings{})
+
+func init() {
+	environment.RegisterEnvironmentMetaDataFlags(envMetaDataStrings)
+}
+
+const (
+	agentRuntime         = 1 * time.Second
+	agentConfigLocalPath = "agent_configs/minimum_config.json"
+	agentConfigPath      = "/opt/aws/amazon-cloudwatch-agent/bin/config.json"
+	agentConfigCopiedDir = "/opt/aws/amazon-cloudwatch-agent/etc/amazon-cloudwatch-agent.d"
+	translatedTomlPath   = "/opt/aws/amazon-cloudwatch-agent/etc/amazon-cloudwatch-agent.toml"
+)
+
+func TestFilePermissions(t *testing.T) {
+	common.CopyFile(agentConfigLocalPath, agentConfigPath)
+	err := common.StartAgent(agentConfigPath, false)
+	if err != nil {
+		log.Printf("Agent failed to start due to err=%v\n", err)
+	}
+	time.Sleep(agentRuntime)
+	common.StopAgent()
+	groupResult := createGroupResult()
+	for _, testResult := range groupResult.TestResults {
+		assert.Equal(t, testResult.Status, status.SUCCESSFUL)
+	}
+	err = common.DeleteFile(agentConfigPath)
+	if err != nil {
+		log.Printf("Failed to delete config file; err=%v\n", err)
+	}
+}
+
+func createGroupResult() status.TestGroupResult {
+	testResults := make([]status.TestResult, 3)
+	testGroupResult := status.TestGroupResult{
+		Name:        "FilePermissions",
+		TestResults: testResults,
+	}
+	testResults[0] = onlyUserCanWriteToFilepath("root", agentConfigPath)
+	testResults[1] = onlyUserCanWriteToFilepath("cwagent", agentConfigCopiedDir)
+	testResults[2] = onlyUserCanWriteToFilepath("cwagent", translatedTomlPath)
+	return testGroupResult
+}
+
+func onlyUserCanWriteToFilepath(user, filepath string) status.TestResult {
+	log.Printf("validating only user=%v can write to filepath=%v", user, filepath)
+	testResult := status.TestResult{
+		Name:   filepath,
+		Status: status.FAILED,
+	}
+
+	hasExpectedOwner, err := fileHasOwnerAndGroup(filepath, user, user)
+	if err != nil || !hasExpectedOwner {
+		return testResult
+	}
+
+	hasExpectedPermission, err := fileHasPermission(filepath, filesystem.OwnerWrite, true)
+	if err != nil || !hasExpectedPermission {
+		return testResult
+	}
+
+	hasExpectedPermission, err = fileHasPermission(filepath, filesystem.AnyoneWrite, false)
+	if err != nil || !hasExpectedPermission {
+		return testResult
+	}
+	log.Printf("SUCCESS: only user=%v can write to filepath=%v", user, filepath)
+	testResult.Status = status.SUCCESSFUL
+	return testResult
+}
+
+func fileHasOwnerAndGroup(filepath, expectedOwner, expectedGroup string) (bool, error) {
+	owner, err := filesystem.GetFileOwnerUserName(filepath)
+	if err != nil {
+		err := fmt.Errorf("FAILED fileHasOwnerAndGroup(); error=%v", err)
+		log.Println(err)
+		return false, err
```

Comment:
I think it's important to log an error if the GetFileOwnerUserName() or GetFileGroupName() fails with error, but yeah it's a good callout that we don't actually use that error anywhere. I'll add some logic for it

Reply:
After some thinking, `onlyUserCanWriteToFilepath()` should not care whether `fileHasOwnerAndGroup()` or `fileHasPermission()` fails due to internal error or incorrect values. However, I think we should still keep the current format because it logs the information needed for debugging.

Reply:
Sure, I'm just saying you probably don't need the boolean since an error can be treated as a failure.